src/main/java/uk/gov/hmcts/ccd/domain/model/common/CatalogueResponse.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.ccd.domain.model.common;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+@JsonDeserialize(builder = CatalogueResponse.Builder.class)
+public class CatalogueResponse<T> {
+
+    private final String code;
+    private final String message;
+    private final T details;
```

Comment:
how come this has been modeled as a generic? isn't a bit overcomplicating things?
If details is dynamic in structure, we could model it as a Map<String, JsonNode> same way as we do for the `data` field of the Case payload.